core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -768,6 +768,19 @@ public static String concat(String s0, String s1) {
     return s0 + s1;
   }
 
+  /** Return null only when both s0 and s1 is null,
+   * otherwise null is treated as empty string. */
+  public static String concatWithNull(String s0, String s1) {
```

Comment:
```suggestion
  public static @Nullable String concatWithNull(String s0, String s1) {
```
in case it can return `null`

Reply:
Done.